### ISSUE TITLE
Refactor template.rb so processors can be "registered"

### ISSUE THREAD

**Issue by stadelmanma** [stadelmanma/sablon#16]
This would work like the content module where document processors can be registered and given a pattern and precedence.  Using my current template.rb setup it would look something like: 

```ruby
register_processor(%r{word/document.xml}, Processor::Document, 0)
register_processor(%r{word/(?:header|footer)\d*\.xml}, Processor::Document, 100)
register_processor(%r{word/footnotes\.xml}, Processor::Document, 200)
register_processor(%r{word/endnotes\.xml}, Processor::Document, 300)
register_processor(%r{word/numbering.xml}, Processor::Numbering, 400)
register_processor(/\[Content_Types\].xml/, Processor::ContentType, 500)
```
I don't know if this is the best approach for files like Numbering and ContentType because those might be better served in a more dynamic fashion. For example, numbering is only messed with when lists are added and never formally processed. The same logic could be applied to *.rels files and content types with some refactoring. 

The main reasoning behind this is it will allow the end user to extend sablon with new processors much easier than they are able to now. Instead of monkey patching we can have a more formal API that while isn't super easy to access is still workable.

**Comment by stadelmanma** [comment]
I don't think I need to use a priority, enforcing some kind of order dependency reintroduces the fragility being fixed by #17 . Done correctly the DOM should be able to manage any "order depend" actions such as the creation of a footnote prior to it's reference being generated since we need the ID.

**Comment by stadelmanma** [comment]
Fixed by https://github.com/senny/sablon/pull/86
